dashboard/utils_test.py

```diff
@@ -0,0 +1,396 @@
+        )
+        # this is a verified enrollment from edx
+        assert mmtrack.is_enrolled_mmtrack("course-v1:edX+DemoX+Demo_Course") is True
+        # this is a audit enrollment from edx
+        assert mmtrack.is_enrolled_mmtrack("course-v1:MITx+8.MechCX+2014_T1") is False
+
+    def test_is_enrolled_mmtrack_fa(self):
+        """
+        Tests for the is_enrolled_mmtrack method in case financial aid is available
+        """
+        course_id = "course-v1:odl+FOO101+CR-FALL15"
+
+        # before paying
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program_financial_aid,
+            enrollments=self.enrollments,
+            current_grades=self.current_grades,
+            certificates=self.certificates
+        )
+        assert mmtrack.is_enrolled_mmtrack(course_id) is False
+
+        # after paying
+        self.pay_for_fa_course(course_id)
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program_financial_aid,
+            enrollments=self.enrollments,
+            current_grades=self.current_grades,
+            certificates=self.certificates
+        )
+        assert mmtrack.is_enrolled_mmtrack(course_id) is True
+
+    def test_has_passed_course_normal(self):
+        """
+        Test for has_passed_course method in case of a normal program
+        """
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program,
+            enrollments=self.enrollments,
+            current_grades=self.current_grades,
+            certificates=self.certificates
+        )
+        # this is a verified enrollment and a verified certificate from edx
+        assert mmtrack.has_passed_course("course-v1:edX+DemoX+Demo_Course") is True
+        # this is a audit enrollment and a non verified certificate from edx
+        assert mmtrack.has_passed_course("course-v1:MITx+8.MechCX+2014_T1") is False
+
+    def test_has_passed_course_fa(self):
+        """
+        Test for has_passed_course method in case of a financial aid program
+        """
+        course_id = "course-v1:odl+FOO101+CR-FALL15"
+        self.pay_for_fa_course(course_id)
+        mmtrack = MMTrack(
+            user=self.user,
+            program=self.program_financial_aid,
+            enrollments=self.enrollments,
+            current_grades=self.current_grades,
+            certificates=self.certificates
+        )
+        assert mmtrack.has_passed_course(course_id) is True
+
+        # move the end date in the future
+        self.crun_fa.end_date = datetime.now(pytz.utc) + timedelta(weeks=1)
+        self.crun_fa.save()
+        assert mmtrack.has_passed_course(course_id) is False
+
+        # remove the end date
+        self.crun_fa.end_date = None
+        self.crun_fa.save()
+        with self.assertRaises(ImproperlyConfigured):
+            mmtrack.has_passed_course(course_id)
+
+        # restore the end date
+        self.crun_fa.end_date = self.end_date
```

Comment:
We shouldn't need this code, `setUpTestData` will roll back the transaction after the test completes